Yank/tests/test_utils.py

```diff
@@ -84,32 +89,35 @@ def test_expand_tree():
               {'simple': {'scalar': 1, 'vector': 4, 'nested': {'leaf': ['d', 'e'], 'combleaf': 'a'}}},
               {'simple': {'scalar': 1, 'vector': 4, 'nested': {'leaf': ['d', 'e'], 'combleaf': 'b'}}},
               {'simple': {'scalar': 1, 'vector': 4, 'nested': {'leaf': ['d', 'e'], 'combleaf': 'c'}}}]
-    assert result == [exp for exp in simple_tree]
+    assert all(exp in result for exp in simple_tree)
 
-    # Test named_combinations generator
-    expected_names = set(['2_a', '3_a', '4_a', '2_b', '3_b', '4_b', '2_c', '3_c', '4_c'])
-    assert expected_names == set([name for name, _ in simple_tree.named_combinations(
-                                                       separator='_', max_name_length=3)])
+    # Test named_combinations generator using either order to account for generator randomness
+    expected_names_A = set(['2_a', '3_a', '4_a', '2_b', '3_b', '4_b', '2_c', '3_c', '4_c'])
+    expected_names_B = set(['a_2', 'a_3', 'a_4', 'b_2', 'b_3', 'b_4', 'c_2', 'c_3', 'c_4'])
+    expected_names = [expected_names_A, expected_names_B]
+    assert set([name for name, _ in simple_tree.named_combinations(separator='_', max_name_length=3)]) in expected_names
```

Comment:
I don't think sorting the leaf itself maters, the problem is the order in which the leaves themselves are returned. Is it possible to tell yaml to use only ordered dictionaries for its construction (I can see that you kind of can in the `yamlbuild.py` file.) Or should we come up with a solution to resume that does not depend on the order of the objects?